lldb/source/Symbol/SaveCoreOptions.cpp

```diff
@@ -101,6 +103,18 @@ bool SaveCoreOptions::ShouldThreadBeSaved(lldb::tid_t tid) const {
   return m_threads_to_save.count(tid) > 0;
 }
 
+bool SaveCoreOptions::HasSpecifiedThreads() const {
+  return !m_threads_to_save.empty();
+}
+
+void SaveCoreOptions::AddMemoryRegionToSave(const lldb_private::MemoryRegionInfo &region) {
+  m_regions_to_save.Insert(region.GetRange(), /*combine=*/true);
```

Comment:
Does this keep things sorted and merge ranges? Or do we post process these ranges somewhere? At some point we will need to iterate over the ranges to save and break them up according to permissions.

Reply:
Insert inserts into a sorted order. I think for the SaveCore frontend we shouldn't worry about permissions and they should instead be split in the process layer.